src/frontend/src/players/parsers/mp4.ts

```diff
@@ -0,0 +1,91 @@
+import { mp4 } from '@taktik/mux.js'
+import { getLogger } from '../../logging/loggers'
+import FfmpegParser from './abstract'
+import { OutputParserResponse } from './types'
+
+class Mp4Parser extends FfmpegParser {
+  protected log = getLogger('mp4 parser')
+  private initSegment?: Buffer
+  private initSegmentComplete = false
+
+  private appendToInitSegment(chunk: Buffer): void {
+    this.initSegment = this.initSegment ? Buffer.concat([this.initSegment, chunk]) : chunk
+  }
+
+  /**
+   * From a chunk of data, attempt to find a "moof" box
+   * This box type announces the beginning of a new data segment
+   * This allows to structure forwarded data to somehthing that can be handled easily
+   * @param chunk 
+   * @returns Data to be added to the inner buffer; and whether a segment has been completed
+   */
+  private parse(chunk: Buffer) {
+    let data = chunk
+
+    // Attempt to find a "moof" box in the chunk
+    // if it is found, it means we reached the end of the previous box's data
+    const [moof] = mp4.tools.findBox(chunk, ['moof'])
+
+    if (!this.initSegmentComplete && moof) {
+      // receiving a "moof" means that the initialization segment is complete
+      this.initSegmentComplete = true
+
+      /**
+       * The call to "findBox" above returns the requested boxes' content if found, without their header
```

Comment:
```suggestion
       * The call to "findBox" above returns the requested boxe's content if found, without their header
```

Reply:
"boxes" is plural here ("findBox" receives an array of box name), it is written as intended